astropy/stats/sigma_clipping.py

```diff
@@ -180,6 +180,7 @@ def sigma_clipped_stats(data, mask=None, mask_val=None, sigma=3.0, iters=None):
         image.
     """
 
+    data = np.asanyarray(data)
```

Comment:
Yes, it's necessary if a `mask` or `mask_val` is input when `data` is a list.  The mask indexing would then fail:

https://github.com/larrybradley/astropy/blob/stats/astropy/stats/sigma_clipping.py#L187
https://github.com/larrybradley/astropy/blob/stats/astropy/stats/sigma_clipping.py#L190

I added tests for these cases.